### DISCUSSION TITLE
Using Pandoc to convert a Latex document that contains unsupported language code (Dart)

### DISCUSSION THREAD

**Discussion by minoush82** [jgm/pandoc#10287] [Q&A] [answered]
I have the following Latex document (`example.tex`).
```
\documentclass[a4paper]{report}
\usepackage{minted}
\begin{document}
	\begin{minted}[linenos, bgcolor=yellow, autogobble, tabsize=4]{dart}
		void main()
		{
			print("I need help!\n");
		}
	\end{minted}

	\begin{minted}[linenos, bgcolor=yellow, autogobble, tabsize=4]{dart}
		// main
		void main()
		{
			print("I need help!\n");
		}
	\end{minted}
\end{document}
```

I am using the following command to get the HTML output (`example.html)`:
`pandoc -s -N --syntax-definition=dart.xml -f latex -t html example.tex -o example.html`

Where `dart.xml` is this one: https://github.com/KDE/syntax-highlighting/blob/master/data/syntax/dart.xml

I get the following warning for each code snippet containing a comment:
`IncludeRules in Dart requires undefined context ##Comments`

The result is that the code snippets containing comments are not syntax-highlighted. The following is what the generated HTML looks like in a browser. As can be seen, the code snippet with comments has no colors.
![example_html](https://github.com/user-attachments/assets/3aa9c7f2-3061-4593-8949-215469c5f6f6)

Any thoughts?

**Comment by minoush82** [comment]
I think the file: https://github.com/KDE/syntax-highlighting/blob/master/data/syntax/dart.xml references other files such as https://github.com/KDE/syntax-highlighting/blob/master/data/syntax/comments.xml using `IncludeRules`.
If this is right, is there a way to include all the necessary XMLs using `--syntax-definition`?

**Reply by jgm** [reply]
Try just putting comments.xml in the working directory, or in the directory containing dart.xml ?

**Reply by minoush82** [reply]
This does not work unfortunately.

**Reply by jgm** [reply, marked as answer]
What if you use `--syntax-definition` twice with both dart.xml and comments.xml, does that work?

**Reply by minoush82** [reply]
Yes yes yes! That works! Thank you very much.

**Comment by minoush82** [comment]
The solution to my problem is to use the following command (thanks to @jgm ):
`pandoc -s -N --syntax-definition=dart.xml --syntax-definition=comments.xml --syntax-definition=spdx-comments.xml -f latex -t html example.tex -o example.html`
